vmdb/app/models/miq_queue.rb

```diff
@@ -1,6 +1,28 @@
 require 'timeout'
 require 'digest'
 
+# Message Bus entry to run a method on any server
+# segmentation:
+#   region
+#      this is partitioned by database, so no need to reference)
+#   zone
+#     this is a portion of the database, most messages are performed in the region of the requester
+#     put: default to myzone. pass in nil to be performed in any zone
+#     get: uses [nil, zone] to fetch my zone and an or for any zone
+#   role
+#     typically nil
+#     put: default to nil
+#     get: use [nil, role] to fetch for my role or any role
+#   queue_name
+#     put: default to "generic"
+#     get: use nil,
+#
+
+# TODO: give justification for all fields
+# TODO: pull priority class out int miq_priority?
+# TODO: add priority scope, and task_id - use AR for everything else
+# TODO: test putting in passing in a zone
+
```

Comment:
This needs a little cleanup.

- I hesitate calling it a Message Bus, because it's not quite one.  Maybe Message Queue.
- Just remove the word segmentation.  I don't understand what that means.
- Don't understand the comment on region where you say "so no need to reference".  It's not even a column. (plus there's an unneeded `)`)
- Zone definition is wrong.  Should say that a zone is a subset of the miq_servers in this region, thus setting zone narrows the scope of the message to only those servers (or something like that).
- Zone put should say "defaults to the zone of the server which calls the put". Also, if you're going to write a sentence, then the next sentence needs a capital letter.
- Zone get is just confusing and gramatically incorrect.
- Role should say "A role defines a set of operations performed by a server".  "typically nil" I think is actually incorrect
- Role put maybe say "default is nil, allowing it to be performed on any server"
- Role get confuses me I think because it should say "my roles" (there are multiple).  Plus the Array notation is confusing in prose.  Not sure what you're trying to convey with it.
- queue_name get has a trailing comma.  Not sure why it says "use nil" either...don't know what that means.
- Typo in TODO 2 int -> into